src/services/pfs/clients/FileSet.ts

```diff
@@ -0,0 +1,49 @@
+import {FileClient, FileClientConstructorArgs} from '../lib/FileClient';
+
+export class FileSet extends FileClient<string> {
+  fileSetId: string | undefined;
+  constructor({
+    pachdAddress,
+    channelCredentials,
+    credentialMetadata,
+    plugins = [],
+  }: FileClientConstructorArgs) {
+    super({
+      pachdAddress,
+      channelCredentials,
+      credentialMetadata,
+      plugins,
+    });
+    const onCallObservers = plugins.flatMap((p) =>
+      p.onCall ? [p.onCall] : [],
+    );
+    const onCompleteObservers = plugins.flatMap((p) =>
+      p.onCompleted ? [p.onCompleted] : [],
+    );
+    const onErrorObservers = plugins.flatMap((p) =>
+      p.onError ? [p.onError] : [],
+    );
```

Comment:
(nit): Seems like we could create some sort of helper function like...

```
const { onCallObservers, onCompleteObservers, onErrorObservers } = deriveObserversFromPlugins(plugins);
```

Not necessary for this PR though, as I know we have this in a ton of places